packages/docprovider/src/ydrive.ts

```diff
@@ -31,55 +37,43 @@ export interface IForkProvider {
   format: string;
 }
 
-/**
- * A Collaborative implementation for an `IDrive`, talking to the
- * server using the Jupyter REST API and a WebSocket connection.
- */
-export class YDrive extends Drive implements ICollaborativeDrive {
-  /**
-   * Construct a new drive object.
-   *
-   * @param user - The user manager to add the identity to the awareness of documents.
-   */
-  constructor(
-    user: User.IManager,
-    translator: TranslationBundle,
-    globalAwareness: Awareness | null
-  ) {
-    super({ name: 'RTC' });
-    this._user = user;
-    this._trans = translator;
-    this._globalAwareness = globalAwareness;
-    this._providers = new Map<string, WebSocketProvider>();
+namespace RtcContentProvider {
+  export interface IOptions extends RestContentProvider.IOptions {
+    user: User.IManager;
+    trans: TranslationBundle;
+    globalAwareness: Awareness | null;
+  }
+}
 
+export class RtcContentProvider
+  extends RestContentProvider
+  implements IContentProvider
+{
+  constructor(options: RtcContentProvider.IOptions) {
+    super(options);
+    this._user = options.user;
+    this._trans = options.trans;
+    this._globalAwareness = options.globalAwareness;
+    this._serverSettings = options.serverSettings;
     this.sharedModelFactory = new SharedModelFactory(this._onCreate);
+    this._providers = new Map<string, WebSocketProvider>();
+    /*
     super.fileChanged.connect((_, change) => {
       // pass through any events from the Drive superclass
       this._ydriveFileChanged.emit(change);
     });
+    */
```

Comment:
I am pretty sure this is no longer needed, but wanted to double check before removal:

```suggestion
```

Reply:
I'm not sure, it looks like you added this change in #337?

Reply:
Yup, I went ahead and removed it, I believe it is not needed as the default drive will emit these.